### PR TITLE
IA-2053 : workflow version changes/follow-up order

### PR DESCRIPTION
**Pull request by beygorghor**
On a workflow details page, follow-ups are not sorted by order and changes are nor not sortable.

Related JIRA tickets : IA-2053

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- change workflow detail api to order follow-ups by order field
- moved changes result from detail cal to a dedicated one, allowing ordering of results
- plug new api features to front-end

## How to test

- Edit a existing non draft workflow version with multiple changes and follow-ups 
- follow-ups should be displayed ordered on order field
- you should be able to change order of changes by clicking in a head cell

## Print screen / video

https://user-images.githubusercontent.com/12494624/232487806-c0baefe8-f81a-4607-8482-96c4e2cb03d2.mov

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/workflows/details.tsx [hunk 1]
```diff
@@ -97,6 +104,13 @@ export const Details: FunctionComponent<Props> = ({ router }) => {
         data?: WorkflowVersionDetail;
         isLoading: boolean;
     } = useGetWorkflowVersion(versionId);
+    const {
+        data: changes,
+    }: // isLoading: isLoadingChanges,
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/workflows/details.tsx]: comment

File: iaso/api/workflows/changes.py [hunk 2]
```diff
@@ -37,11 +39,21 @@ class WorkflowChangeViewSet(viewsets.ViewSet):
...
     serializer_class = ser.WorkflowChangeSerializer
+    http_method_names = ["get", "post", "delete", "put"]
 
     @swagger_auto_schema(
         manual_parameters=[version_id_param],
         request_body=ser.WorkflowChangeCreateSerializer,
     )
+    def get_queryset(self):
+        orders = self.request.query_params.get("order", "updated_at").split(",")
+        queryset = WorkflowChange.objects
```
Comment (quang-le) [inline iaso/api/workflows/changes.py:50]: shouldn't that be `objects.all()`?

**Review by quang-le — APPROVED** [review]

File: iaso/api/workflows/changes.py (hunk 2 above)
Comment (beygorghor) [reply iaso/api/workflows/changes.py:50]: i think the all is required if you want to parse it, not to build a `queryset`

**Comment by beygorghor** [comment]
This is breaking cypress tests

File: iaso/api/workflows/serializers.py [hunk 3]
```diff
@@ -169,6 +169,9 @@ class Meta:
         model = WorkflowFollowup
         fields = ["id", "order", "condition", "forms", "created_at", "updated_at"]
 
+    def get_queryset(self):
```
Comment (mathvdh) [inline iaso/api/workflows/serializers.py]: I think we can remove this as it was not working ..

**Review by mathvdh — CHANGES_REQUESTED** [review]
I have put a comment with 3 lines to be removed otherwise it seems good to me

**Pull request merged**